checkov/terraform/graph_builder/local_graph.py

```diff
@@ -320,11 +407,26 @@ def _connect_module_provider(self) -> None:
                             for e in referenced_vertices:
                                 if self.vertices[e.dest].block_type == BlockType.PROVIDER:
                                     for resource in resources:
-                                        # connect resource to provider
+                                        self._add_provider_address_to_resource(e, resource)
                                         self.create_edge(resource, e.dest, e.label)
                 except Exception as e:
                     logging.warning(f"Failed in connecting module resources to provider due to {e}")
 
+    @staticmethod
+    def _get_resource_name_and_type_from_name(name: str) -> tuple[str, str]:
+        resource_name, resource_type = '', ''
+        split_name = name.split('.')
+        if len(split_name) >= 2:
```

Comment:
this will not work if you have for example `.` inside a foreach key. 
Instead you can use this function `get_sanitized_terraform_resource_id` (or a variation of it) from [here](https://github.com/bridgecrewio/checkov/blob/c248a676005c00ea88d5b3695f0b0d656468efa0/checkov/terraform/graph_builder/foreach/utils.py)

Reply:
Remove, already implemented logic in another func